soh/soh/Enhancements/randomizer/3drando/hints.cpp

```diff
@@ -963,7 +964,7 @@ static void DistributeHints(std::vector<uint8_t>& selected, size_t stoneCount, s
     for (uint8_t distribution = 0; distribution < distTable.size(); distribution++){
       currentWeight -= distTable[distribution].weight;
       if (currentWeight <= 0){
-        if (stoneCount >= distTable[distribution].copies){
+        if (stoneCount >= distTable[distribution].copies || distTable[distribution].copies == 0){
```

Comment:
I assume this is to catch an edge case, but i can't think of any case where this would trigger as it implies that stoneCount is in negative numbers which is an erroneous state in it's own right.

Even then, if copies is zero and the hint type has weight that's also an erroneous state, albeit one that should be accounted for, specifically a hint distribution is declared but will never be assigned stones, so there is no reason to select it instead of letting the stone check fail and the code remove it from consideration.

Reply:
When it fails to place a hint in `PlaceHints` it sets copies to 0 for that distTable entry. This was the only way I had to make it not distribute hints somewhere that it failed to place a hint in the previous cycle.

Reply:
Without this check, it was re-distributing hints back into a distribution that it failed to place hints in earlier.